extensions/amp-video/0.1/test/test-video-cache.js

```diff
@@ -0,0 +1,159 @@
+import {createElementWithAttributes} from '../../../../src/dom';
+import {createExtensionScript} from '../../../../src/service/extension-script';
+import {xhrServiceForTesting} from '../../../../src/service/xhr-impl';
+
+describes.realWin('amp-video cached-sources', {amp: true}, (env) => {
+  let xhrService;
+  let cacheUrlService;
+
+  beforeEach(() => {
+    xhrService = xhrServiceForTesting(env.win);
+    env.sandbox.stub(Services, 'xhrFor').returns(xhrService);
+
+    cacheUrlService = new AmpCacheUrlService();
+    env.sandbox
+      .stub(Services, 'cacheUrlServicePromiseForDoc')
+      .resolves(cacheUrlService);
+    env.win.document.head.appendChild(
+      createExtensionScript(env.win, 'amp-cache-url', '0.1')
+    );
+    env.sandbox
+      .stub(Services, 'documentInfoForDoc')
+      .returns({sourceUrl: 'https://example.com'});
+  });
+
+  describe('select sources', () => {
+    it('should select the source if there is only one source', () => {
+      const videoEl = createVideo([{src: 'video1.mp4'}]);
+      expect(videoCache.selectVideoSource(videoEl)).to.contain('video1.mp4');
+    });
+
+    it('should select the first source if there are similar sources', () => {
+      const videoEl = createVideo([{src: 'video1.mp4'}, {src: 'video2.mp4'}]);
+      expect(videoCache.selectVideoSource(videoEl)).to.contain('video1.mp4');
+    });
+
+    it('should select the mp4 source if there are many sources', () => {
+      const videoEl = createVideo([
+        {src: 'video1.mp4'},
+        {src: 'video2.mp4', type: 'video/mp4'},
+      ]);
+      expect(videoCache.selectVideoSource(videoEl)).to.contain('video2.mp4');
+    });
+  });
+
+  describe('add sources', () => {
+    it('should set the correct attributes on the source added', () => {
+      const videoEl = createVideo([]);
+      videoCache.applySourcesToVideo(videoEl, [
+        {'url': 'video1.mp4', 'bitrate_kbps': 700, type: 'video/mp4'},
+      ]);
+      const addedSource = videoEl.querySelector('source');
+      expect(addedSource.getAttribute('src')).to.equal('video1.mp4');
+      expect(addedSource.getAttribute('data-bitrate')).to.equal('700');
+      expect(addedSource.getAttribute('type')).to.equal('video/mp4');
+    });
+
+    it('should add the sources sorted by bitrate', () => {
+      const videoEl = createVideo([]);
+      videoCache.applySourcesToVideo(videoEl, [
+        {'url': 'video1.mp4', 'bitrate_kbps': 700, type: 'video/mp4'},
+        {'url': 'video2.mp4', 'bitrate_kbps': 2000, type: 'video/mp4'},
+        {'url': 'video3.mp4', 'bitrate_kbps': 1500, type: 'video/mp4'},
+      ]);
+      const addedSources = videoEl.querySelectorAll('source');
+      expect(addedSources[0].getAttribute('data-bitrate')).to.equal('2000');
+      expect(addedSources[1].getAttribute('data-bitrate')).to.equal('1500');
+      expect(addedSources[2].getAttribute('data-bitrate')).to.equal('700');
+    });
+  });
+
+  describe('url forming', () => {
```

Comment:
These tests are better but if you write them this way, they should live in amp-video.
Here, you should call `fetchCachedSources` and keep your assertion on the XHR.